src/qibolab/serialize.py

```diff
@@ -85,17 +85,16 @@ def load_qubits(
 
 
 def _load_pulse(pulse_kwargs, qubit):
-    pulse_type = pulse_kwargs.pop("type")
-    if "coupler" in pulse_kwargs:
-        q = pulse_kwargs.pop("coupler", qubit.name)
-    else:
-        q = pulse_kwargs.pop("qubit", qubit.name)
+    coupler = "coupler" in pulse_kwargs
+    q = pulse_kwargs.pop("coupler" if coupler else "qubit", qubit.name)
 
-    if pulse_type == "dl":
-        return Delay(**pulse_kwargs)
-    if pulse_type == "virtual_z":
+    if "phase" in pulse_kwargs:
         return VirtualZ(**pulse_kwargs, qubit=q)
```

Comment:
Ok, I get your point.

The main motivation for this was that it is the poor-man version of the Pydantic deserialization (in its words, validation), that is pattern matching the fields of the union.

Since for the time being this was enough to identify the `VirtualZ`, I didn't bother to spell out the full identification condition. Hopefully that will be generated automatically before creating a clash with the `"phase"` (otherwise we will schedule it to happen this way).

Reply:
Another solution is the `"kind"` approach that you used to distinguish the different envelopes. However, this would come at the price of having `"kind": "pulse"` everywhere in the runcards, so I would not go for it. 

In any case, I would postpone this until it is really an issue, so I will resolve this point.